### ISSUE TITLE
Color map and biom file with non-exact matching

### ISSUE THREAD

**Issue by mooreryan** [mooreryan/iroki#21]
Biom files should always follow exact matching, but in theory, color map could be regex while biom is exact. 

Quick fix is to only allow `--exact` when providing both biom and color map.

```
/Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/lib/iroki/utils/utils.rb:49:in `=~': type mismatch: String given (TypeError)
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/lib/iroki/utils/utils.rb:49:in `block in add_color_to_leaf_branch'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/lib/iroki/utils/utils.rb:48:in `each'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/lib/iroki/utils/utils.rb:48:in `add_color_to_leaf_branch'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/lib/iroki/main/main.rb:373:in `block in main'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/lib/iroki/main/main.rb:370:in `map'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/lib/iroki/main/main.rb:370:in `main'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/gems/iroki-0.0.32/exe/iroki:92:in `<top (required)>'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/bin/iroki:23:in `load'
	from /Users/moorer/.rvm/gems/ruby-2.3.1@iroki/bin/iroki:23:in `<main>'
```

Files for the error.

[rumino_abund.txt](https://github.com/mooreryan/iroki/files/927378/rumino_abund.txt)
[fh_color_map.txt](https://github.com/mooreryan/iroki/files/927376/fh_color_map.txt)
[samples.upgma.tre.txt](https://github.com/mooreryan/iroki/files/927377/samples.upgma.tre.txt)

**Comment by mooreryan** [comment]
The quick fix was implemented with this commit 69ecf8574db05ef90d58deeb389dfad0534cb6ca

**Comment by mooreryan** [comment]
We could allow regex with biom files but it might get confusing. For example if the pattern in the color map is `f blue` and the user didn't add any tags to the color map file, then it would override every fecal sample in the biom file, which is probably not what they wanted. 

So for now, I'm closing this with only the quick fix mentioned.
